core/src/test/java/org/apache/iceberg/avro/TestSchemaConversions.java

```diff
@@ -370,4 +370,17 @@ public void testFieldDocsArePreserved() {
         Lists.newArrayList(Iterables.transform(origSchema.columns(), Types.NestedField::doc));
     assertThat(fieldDocs).isEqualTo(origFieldDocs);
   }
+
+  @Test
+  public void testVariantConversion() {
+    org.apache.iceberg.Schema schema =
+        new org.apache.iceberg.Schema(required(1, "variantCol", Types.VariantType.get()));
+    org.apache.avro.Schema avroSchema = AvroSchemaUtil.convert(schema.asStruct());
+
+    org.apache.avro.Schema variantSchema = avroSchema.getField("variantCol").schema();
```

Comment:
I think we also need a test that multiple variant columns can be converted in a single Avro record.

Other test cases (like `testPrimitiveTypes`) validate Avro to Iceberg conversion. Do you want to add that as well?

Reply:
For test cases like `testPrimitiveTypes`, right now we have the same dependency  (as BuildAvroProjection) on adding Variant logical type to Avro. I will add that later.

Reply:
I changed the test to test with 2 variant columns now. I think we don't need to keep the test for 1 variant column so I don't keep it here.